actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -112,27 +114,84 @@ private void addCourseDetails(Request request, Map<String, Object> userCoursesRe
+      List<Map<String, Object>> courses =
+          (List<Map<String, Object>>) courseBatchesMap.get(JsonKey.CONTENT);
+      if (CollectionUtils.isNotEmpty(courses)) {
+        courses.forEach(
+            course -> courseBatches.put((String) course.get(JsonKey.IDENTIFIER), course));
+      }
+      ProjectLogger.log(
+          "LearnerStateActor:addCourseDetails: coursesBathces = " + courseBatches,
+          LoggerEnum.INFO.name());
+    }
+    mergeDetailsAndSendCourses(contents, batches, courseBatches);
+  }
+
+  private Map<String, Object> getCourseBatch(
+      List<Map<String, Object>> batches, List<String> requestedFields) {
+    List<String> courseBatchIds =
+        (List<String>)
+            batches
+                .stream()
+                .map(batch -> (String) batch.get(JsonKey.BATCH_ID))
+                .collect(Collectors.toList());
+
+    Map<String, Object> esQueryMap = new HashMap<>();
+    esQueryMap.put(JsonKey.IDENTIFIER, courseBatchIds);
+    SearchDTO dto = new SearchDTO();
+    requestedFields.add(JsonKey.IDENTIFIER);
+    dto.setFields(requestedFields);
+    dto.getAdditionalProperties().put(JsonKey.FILTERS, esQueryMap);
+    return ElasticSearchUtil.complexSearch(
+        dto, ProjectUtil.EsIndex.sunbird.getIndexName(), ProjectUtil.EsType.course.getTypeName());
   }
 
   public void mergeDetailsAndSendCourses(
-      Map<String, Object> coursesContents, List<Map<String, Object>> batches) {
+      Map<String, Object> coursesContents,
+      List<Map<String, Object>> batches,
+      Map<String, Object> courseBatches) {
 
     ProjectLogger.log(
-        "LearnerStateActor:prepareCourseBatchResponse coursesContents =" + coursesContents,
+        "LearnerStateActor:mergeDetailsAndSendCourses coursesContents =" + coursesContents,
         LoggerEnum.DEBUG.name());
 
-    Map<String, Object> contentsByCourseId = new HashMap<>();
-    if (MapUtils.isNotEmpty(coursesContents)) {
-      List<Map<String, Object>> courses =
-          (List<Map<String, Object>>) coursesContents.get(JsonKey.CONTENTS);
-      if (CollectionUtils.isNotEmpty(courses)) {
-        courses.forEach(
-            course -> contentsByCourseId.put((String) course.get(JsonKey.IDENTIFIER), course));
+    if (MapUtils.isNotEmpty(courseBatches)) {
+      ProjectLogger.log(
+          "LearnerStateActor:mergeDetailsAndSendCourses courseBatchContents ="
+              + "for requested data "
+              + courseBatches,
+          LoggerEnum.DEBUG.name());
+    }
+
+    Map<String, Object> contentsByCourseId = getContentAsMap(coursesContents);
+
+    if (MapUtils.isNotEmpty(contentsByCourseId) && MapUtils.isNotEmpty(courseBatches)) {
+      Set<String> batchIds = contentsByCourseId.keySet();
+      for (String batchId : batchIds) {
+        if (courseBatches.containsKey(batchId)) {
+          Map<String, Object> course = (Map<String, Object>) contentsByCourseId.get(batchId);
+          course.put(JsonKey.BATCH, courseBatches.get(batchId));
```

Comment:
Will this return 'id' within batch?

Reply:
yes , it will return. id is required to distinguish between batch objects while appending them into userCourses object.
if in requested parameters "fields" is empty then we will not make a call to elastic search.